### PR TITLE
L3 fixes

### PR DESCRIPTION
**Pull request by npm1**
This PR does the following:
* Remove "= null" from the detail fields, fixing https://github.com/w3c/user-timing/issues/56.
* Replace the SyntaxErrors introduced in L3 with either RangeErrors or TypeErrors, fixing https://github.com/w3c/user-timing/issues/55.
* Make startOrOptions not nullable again, fixing https://github.com/w3c/user-timing/issues/54. The optional dictionaries (or union including dictionary) have an empty dictionary as its default value. Thus, the logic in measure() is changed accordingly because when an optional dictionary is not given, it will be defaulted to the empty dictionary.
* Replace some of the "is present" with "is given" because that is more accurate terminology for optional arguments.

@maxlgu FYI

### CODE REVIEW

**Comment by npm1** [comment]
Tagging @bzbarsky since I cannot add you as a reviewer.

File: index.html [hunk 1]
```diff
@@ -226,16 +230,16 @@ <h2><dfn>measure()</dfn> method</h2>
           <li>
             Compute <var>start time</var> as follows:
             <ol>
-              <li>If <var>startOrMeasureOptions</var> is present and is a <a>PerformanceMeasureOptions</a> object, and if its <a>start</a> member is present, let <var>start time</var> be the value returned by running the <a>convert a mark to a timestamp</a> algorithm passing in <var>startOrMeasureOptions</var>'s <a>start</a>.</li>
+              <li>If <var>startOrMeasureOptions</var> is a <a>PerformanceMeasureOptions</a> object, and if its <a>start</a> member is present, let <var>start time</var> be the value returned by running the <a>convert a mark to a timestamp</a> algorithm passing in <var>startOrMeasureOptions</var>'s <a>start</a>.</li>
```
Comment (yoavweiss) [inline index.html]: Can you link "present" to its definition?

File: index.html [hunk 2]
```diff
@@ -310,23 +314,17 @@ <h3>The <dfn><a>PerformanceMark</a> Constructor</dfn></h3>
             Set <var>entry</var>'s <code>startTime</code> attribute as follows:
             <ol>
               <li>
-                If <var>markOptions</var> is present and its <a>startTime</a> member is present, then:
+                If <var>markOptions</var>'s <a>startTime</a> member is present, then:
```
Comment (yoavweiss) [inline index.html]: Why did you remove the check to see if `markOptions` is given?

File: index.html (hunk 1 above)
Comment (npm1) [reply index.html]: Done

File: index.html (hunk 2 above)
Comment (npm1) [reply index.html]: In https://heycam.github.io/webidl/#idl-operations it says:
"If the type of an argument is a dictionary type or a union type that has a dictionary as one of its flattened member types, and that dictionary type and its ancestors have no required members, and the argument is either the final argument or is followed only by optional arguments, then the argument must be specified as optional. Such arguments are always considered to have a default value of an empty dictionary, unless otherwise specified."
I interpret this as meaning that those arguments are considered to always be given, but not sure about this...

Comment (npm1) [reply index.html]: @heycam do you know if the check for whether this argument is given or not should be done?

Comment (domenic) [reply index.html]: Yep, it's always given, per your reading. I agree it's a little non-obvious. It would be more obvious to require everyone to put `= {}` after it. https://github.com/heycam/webidl/issues/602

Comment (yoavweiss) [reply index.html]: Cool, thanks!

File: index.html [hunk 3]
```diff
@@ -310,23 +314,17 @@ <h3>The <dfn><a>PerformanceMark</a> Constructor</dfn></h3>
             Set <var>entry</var>'s <code>startTime</code> attribute as follows:
             <ol>
               <li>
-                If <var>markOptions</var> is present and its <a>startTime</a> member is present, then:
+                If <var>markOptions</var>'s <a>startTime</a> member is <a data-cite="WEBIDL#dfn-present">present</a>, then:
                 <ol>
-                  <li>If <var>markOptions</var>'s <a>startTime</a> is negative, throw a <a data-cite="WEBIDL#syntaxerror"><code>SyntaxError</code></a>.</li>
+                  <li>If <var>markOptions</var>'s <a>startTime</a> is negative, throw a <a data-cite="WEBIDL#exceptiondef-rangeerror"><code>RangeError</code></a>.</li>
```
Comment (bzbarsky) [inline index.html]: Usually this would be a TypeError in web APIs (e.g. that's what WebIDL uses for [EnforceRange] violations).

File: index.html [hunk 4]
```diff
@@ -364,7 +362,7 @@ <h2>Convert a <var>mark</var> to a <var>timestamp</var></h2>
           <li>
             Otherwise, if <var>mark</var> is a <code>DOMHighResTimeStamp</code>:
             <ol>
-              <li>If <var>mark</var> is negative, throw a <a data-cite="WEBIDL#syntaxerror"><code>SyntaxError</code></a>.</li>
+              <li>If <var>mark</var> is negative, throw a <a data-cite="WEBIDL#exceptiondef-rangeerror"><code>RangeError</code></a>.</li>
```
Comment (bzbarsky) [inline index.html]: Same thing here.

**Comment by npm1** [comment]
Ok I will send a quick PR to replace those RangeErrors with TypeErrors.

File: index.html (hunk 3 above)
Comment (yoavweiss) [reply index.html]: I'm curious - is there a document I could read to better outline which errors should be thrown in which occasions? https://heycam.github.io/webidl/#idl-exceptions is not super descriptive when it comes to when each one of them should be used

Comment (bzbarsky) [reply index.html]: I don't know that we have anything great.  :(
